### DISCUSSION TITLE
Can pybamm achieve low temperature fast charging and preheating?

### DISCUSSION THREAD

**Discussion by RobinBluehub** [pybamm-team/PyBaMM#2869] [Q&A]
Hello Pybamm team, currently I want to study low temperature fast charging. Before fast charging, I want to implement a battery warm-up phase, to the charging temperature I want to set. There are no relevant examples in the forum. Can pybamm achieve battery warm-up?
Best regards！
Robin

**Comment by rtimms** [comment]
You can [pass the temperature as a function of time](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/scripts/DFN_ambient_temperature.py) or give a temperature to steps of an experiment.
